MetaMask/metamask-extension
# Review spending cap screen

## Explanation
Created a new screen - 'Review Spending Cap' screen.


## More Information
* Fixes #15220 


## Screenshots/Screencaps



### Before

https://user-images.githubusercontent.com/92310504/191487193-b08b626c-0b3f-4f82-a055-eebb54828e08.mov




### After




https://user-images.githubusercontent.com/92310504/193078487-4a9da6a2-a4d2-464e-a274-b48fe840c097.mov







## Manual Testing Steps
**BEFORE TESTING** - in your `.metamaskrc` file, set `TOKEN_ALLOWANCE_IMPROVEMENTS=1` and then start the app

1. Open the Test Dapp: https://metamask.github.io/test-dapp/
2. Under the section `Send Tokens` click on `CREATE TOKEN`
3. Wait for the token to be created and then, under the same section, click on `APPROVE TOKENS` or `APPROVE TOKENS WITHOUT GAS` in order to see the newly created screen

To see how the screen looks like when the enhanced gas fee is enabled, go to `Settings (Experimental tab)` and enable it (`Enable enhanced gas fee UI` toggle).
